contents/ruby/Rails.md

```diff
@@ -0,0 +1,157 @@
+
+For example, in order to fetch yesterday's date in Python, one may have to write
+
+```python
+import datetime
+datetime.datetime.now() - datetime.timedelta(days = 1)
+```
+
+The same action in Rails is done using
+```ruby
+1.day.ago
+```
+
+Similarly, Rails also added a new method to access an array element. Instead of following the C convention of using square-brackets to access the second element in the array 'elements' like
+
+```c
+elements[1]
+```
+
+Rails developers can use ordinal representation to access the same element with the command
+
+```ruby
+elements.second
+```
+
+And while this may not reduce the amount one has to type, most Rails enthusiasts agree that it makes the code much more readable, intuitive, and inherently happiness-inducing.
+
+### Great Community of Developers
+
+The Ruby language has a very passionate and extensive community of developers, and there is a vast array of libraries (called 'Gems'), that can be used with your Rails projects. These libraries provide a lot of functionalities, and can be used to add flexibility to Rails projects when required. This may even help negate some of the drawbacks of Rails, which are as follows.
+
+### Drawbacks
+
+#### Lack of Flexibility
+
+As demonstrated above, Rails follows very strict conventions, that make life much easier for developers - or so Rails thinks. It might be true that there are some developers who find Rails' convention very stifling, and, as a result, prefer to use frameworks that allow them more flexibility. Rails will still allow you to do as you please, but since the effort of undo-ing the conventions to follow your own preferences essentially negates any time advantage granted by Rails, you may choose to use a different framework.
+
+#### Performance Time
+
+While newer releases have combatted this problem to a large degree, many of the old releases of Rails tend to be slower than their counterparts. Rails is also used to make server-side applications that send static HTML files, with minimal Ruby code to modify the appearance. Since the current trend is towards making proper and responsive client-side, front-end applications, Rails is a comparatively 'old school' approach to web developement. However, there are many Gems that allow you to combine your Rails back-end with a modern React, Angular, or Vue front-end, thus enabling you to create responsive static apps with Rails, but it has to be noted that the default for Rails does not accomodate such applications, thus giving it a worse performance time as compared to a client-side application.
```

Comment:
FWIW it's possible to use Rails in API-only mode (i.e. no views, just routes/controllers/models) and deploy it together with a front-end framework that interfaces with your Rails backend.

Reply:
Whoops TIL haha. I'll have to read up more on that in that case. Thanks!